src/util/createServerRootMixin.js

```diff
@@ -130,37 +130,21 @@ function augmentInstantSearch(
       .then(() => renderToString(app))
       .then(() => searchOnlyWithDerivedHelpers(helper))
       .then(() => {
-        const results = {};
-        walkIndex(renderedComponent.instantsearch.mainIndex, widget => {
-          results[widget.getIndexId()] = widget.getResults();
+        resultsState = {};
+        walkIndex(instance.mainIndex, widget => {
+          const { _state, _rawResults } = widget.getResults();
```

Comment:
same as the react implementation, but I wonder if this instead was:

```js
const state = widget.getWidgetSearchParameters(new SearchParameters(), { uiState: search._initialUiState[widget.getIndexId()] }
```

if we then still have the wrong parameters available `helper.state`, or if that's correct then?

Reply:
Wdym?

Reply:
instead of taking the search parameters from the results (as is done now, and in React InstantSearch too), we could retrieve the "local" index search parameters by using getWidgetSearchParameters of each index, although it likely still would not work completely out of the box, as by default it's recursive. If that could be avoided, there would be a way to get the correct search parameters that can be applied to helper.state without issue

Reply:
I don't see how that's easier than using `getResults()`?

Reply:
I didn't mean easier, I meant more correct. I was trying to find a way to get only the "local index search parameters", but indeed what I was suggesting wasn't correct, so I guess it doesn't matter much